prepare-translations-for-npm/prepare-translations.js

```
@@ -0,0 +1,57 @@
+const chalk = require('chalk'),
+	fs = require('fs');
+
+const langs = ['ar', 'cy', 'da', 'de', 'es', 'es-es', 'fr', 'fr-fr', 'ja', 'ko', 'nl', 'pt', 'sv', 'tr', 'zh-tw', 'zh', 'zebras'];
+
+function _parseFile(filePath) {
+	const file = fs.readFileSync(filePath).toString();
+	const firstBit = file.split('default ')[1];
+	const stringContent = firstBit.substring(0, firstBit.lastIndexOf(';'));
+	return JSON.parse(stringContent);
+}
+
+function _writeChanges(langTermsPath, fileName, content) {
+	const json = JSON.stringify(content, null, '\t');
+	const fileContent = `/* eslint quotes: 0 */
+export default ${json};
+`;
+	fs.writeFileSync(`${langTermsPath}/${fileName}.js`, fileContent, 'utf8');
+}
+
+function prepare(langTermsPath) {
+	const englishTranslations = _parseFile(`${langTermsPath}/en.js`);
+	const englishKeys = Object.keys(englishTranslations);
+
+	langs.forEach((lang) => {
+		const filePath = `${langTermsPath}/${lang}.js`;
+
+		if (!fs.existsSync(filePath)) return;
+		
+		let changes = false;
+		const translations = _parseFile(filePath);
+		englishKeys.forEach((key) => {
+			const translatedValue = translations[key];
+			if (translatedValue === undefined || translatedValue === '') {
+				changes = true;
+				translations[key] = englishTranslations[key];
+			}
+		});
+
+		if (changes) {
+			const ordered = {};
+			Object.keys(translations).sort().forEach((key) => {
+				ordered[key] = translations[key];
+			});
+			_writeChanges(langTermsPath, lang, ordered);
+		}
+	});
+}
+
+try {
+	prepare(process.env['LANG_PATH']);

```

This is effectively the same as the existing CLI script, except we take the lang path in as a parameter and ignore any missing translation files. I think if we want to throw some kind of warning for missing translation files, it's maybe more appropriate for that to happen in some CI-based verify step (as discussed in Slack), but I'm open to opinions otherwise.

---

That's fair - if we did want to try and handle the missing language case in a more automatic way, I think it would need to happen in a different job anyway, because it would need to commit a couple changes (the new file, the serge config file, and we'd need to share the localize mixin somehow). An action that verifies could be the beginning of that.